sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -129,13 +129,20 @@ class FileStreamSource(
     val files = metadataLog.get(Some(startId + 1), Some(endId)).flatMap(_._2)
     logInfo(s"Processing ${files.length} files from ${startId + 1}:$endId")
     logTrace(s"Files are:\n\t" + files.mkString("\n\t"))
+    val newOptions = if (!SparkHadoopUtil.get.isGlobPath(new Path(path))) {
```

Comment:
I recommend putting this code into the initialization code at the top of this file that sets `qualifiedBasePath` (currently lines 47-50). That way all the code that interprets the meaning of the `path` parameter` will be in one place.